pulser-core/pulser/pulse.py

```diff
@@ -263,6 +265,26 @@ def __repr__(self) -> str:
             + f"post_phase_shift={self.post_phase_shift:.3g})"
         )
 
+    def __eq__(self, other: Any) -> bool:
+        if type(other) is not type(self):
+            return False
+
+        def check_phase_eq(phase1: float, phase2: float) -> np.bool_:
+            # Comparing with an offset ensures we don't fail just because
+            # we are very close to the wraping point
+            return np.isclose(phase1, phase2, atol=1e-6) or np.isclose(
+                (phase1 + 1) % (2 * np.pi),
+                (phase2 + 1) % (2 * np.pi),
+                atol=PHASE_PRECISION,
+            )
```

Comment:
Well this only shifts the phase value at which the issue arrises no ? from $2\pi$  to $2\pi -1$

Reply:
I am not sure I understand... If you have phase1 at 2pi-1 - 0.0000001 and phase2 at 2pi-1+0.0000001  they will not be considered as close right ?

Reply:
Yes, but they will be considered close on the first `np.isclose()` (note that all phases are already stored mod 2pi)

Reply:
Aaah damn now I get it sorry I didn't see the "or" 🙃 Thanks !